all_test.go

```diff
@@ -135,6 +135,30 @@ func TestCopy(t *testing.T) {
 		err = Copy(src, dest, opt)
 		Expect(t, err).ToBe(nil)
 	})
+	When(t, "fs copy should not write outside fs", func(t *testing.T) {
+		subdir := t.TempDir()
+		subfs := os.DirFS(subdir)
+
+		// Copy assets to subdir so that they're available for fs.
+		src := "test/data/case18/assets"
+		err := Copy(src, filepath.Join(subdir, src))
+		Expect(t, err).ToBe(nil)
+
+		// Define a new destination for the assets in the subdir so they are reachable.
+		dest := "test/data.copy/case18/assets"
+		err = Copy(src, dest, Options{FS: subfs})
+		Expect(t, err).ToBe(nil)
+
+		_, err = os.Stat(filepath.Join(subdir, dest))
+		Expect(t, err).ToBe(nil)
+	})
+	When(t, "fs copy relative when fs is root", func(t *testing.T) {
```

Comment:
I could roll the ball either way whether or not this is an error or a feature. Here `Copy` is not respecting the working directory when using `FS`, if it was the test would pass.

OTOH, when providing the `FS` you might not want this behavior.